.github/workflows/pull-request-checks.yaml

```diff
@@ -52,22 +51,21 @@ jobs:
           # user input
           DEBIAN_FRONTEND: noninteractive
         run: |
-          sudo apt-get install -yq cmake ninja-build gcc g++ maven flex bison libxml2-utils cpanminus dpkg-dev
-          cpanm Thread::Pool::Simple
```

Comment:
Why is this perl module being removed? It is required for the multi threading functionality of `test.pl`. This should currently be enabled via the `TESTPL_JOBS` environment variable.

Reply:
Yes, sorry, I overlooked it - please don't delete this, this is actually helpful.

Reply:
Except ... we never set TESTPL_JOBS anywhere? I indeed was pondering to set this variable, but then we had been successful on Travis in the past with the use of GNU parallel, hence I opted for using the `test-parallel` target instead. But I'm fine either way. FWIW, the time for running the regression tests was almost exactly halved.

Reply:
`TESTPL_JOBS` at least for this particular action, is being set in line 71 of this file as an environment variable, to my understanding.

Reply:
I think the `test.pl` level of multi threading is included in this `.yaml` but isn't actually working correctly. Looking at the CI log from another PR there are existing warnings.
<details>
<summary> `cpanm` output </summary>

```
!
! Can't write to /usr/local/share/perl/5.30.0 and /usr/local/bin: Installing modules to /home/runner/perl5
! To turn off this warning, you have to do one of the following:
!   - run me as a root or with --sudo option (to install to /usr/local/share/perl/5.30.0 and /usr/local/bin)
!   - Configure local::lib in your existing shell to set PERL_MM_OPT etc.
!   - Install local::lib by running the following commands
!
!         cpanm --local-lib=~/perl5 local::lib && eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
!
--> Working on Thread::Pool::Simple
Fetching http://www.cpan.org/authors/id/J/JW/JWU/Thread-Pool-Simple/Thread-Pool-Simple-0.25.tar.gz ... OK
Configuring Thread-Pool-Simple-0.25 ... OK
Building and testing Thread-Pool-Simple-0.25 ... OK
Successfully installed Thread-Pool-Simple-0.25
```

</details>  


<details>
<summary> `test.pl` output </summary>

```
1: Jobs set but thread pool module not found,
1: install with 'cpan -i Thread::Pool::Simple'
```

</details> 

I am fine with removing this, as it isn't currently working. However I would prefer the removal to be in a separate commit with the "because it doesn't work" explanation. That way if we ever want to re-instate it we can revert that commit only.

Reply:
Thanks, will move this to a commit of its own!